sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample5_GetSettingIfChanged.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using Azure.Core.Testing;
+using NUnit.Framework;
+
+namespace Azure.Data.AppConfiguration.Samples
+{
+    [LiveOnly]
+    public partial class ConfigurationSamples
+    {
+        [Test]
+        /*
+         * This sample ilustrates how to get a setting from the configuration
+         * store only if the version in the configuration store is different
+         * from the one held by your client application, as determined by whether
+         * the setting ETags match.  Getting a setting only if it has changed
+         * allows you to avoid downloading a setting if your client application
+         * is already holding the latest value, which saves cost and bandwidth.
+         */
+        public void GetSettingIfChanged()
+        {
+            // Retrieve the connection string from the environment.
+            // The connection string is available from the App Configuration Access Keys view in the Azure Portal.
+            var connectionString = Environment.GetEnvironmentVariable("APPCONFIGURATION_CONNECTION_STRING");
+
+            // Instantiate a client that will be used to call the service.
+            var client = new ConfigurationClient(connectionString);
+
+            // Create a Configuration Setting to be stored in the Configuration Store.
+            ConfigurationSetting setting = new ConfigurationSetting("some_key", "initial_value");
+
+            // Add the setting to the Configuration Store.
+            ConfigurationSetting settingV1 = client.SetConfigurationSetting(setting);
+
+            // Download the setting only if it's changed.
+            Response<ConfigurationSetting> response = client.GetConfigurationSetting(settingV1, onlyIfChanged: true);
+
+            // This ConfigurationSetting object will hold the most up-to-date version of the setting.
+            ConfigurationSetting latestSetting = null;
+
+            // Check to see whether the setting was changed in the configuration store before accessing its value.
+            // If response.Value is accessed when no value was returned, an exception will be thrown.
+            Debug.WriteLine($"Received a response code of {response.GetRawResponse().Status}");
+            if (response.GetRawResponse().Status == 304)
```

Comment:
Could someone just initial `latestSetting` to `settingV1` and overwrite it if changed, or is the idea here to show how status code 304 works in this scenario?

Reply:
Yes, that would totally work and be cleaner, but the key point here is that if you get a 304 and then reference response.Value, it will throw an exception, and you can avoid that by checking for a 304 status code.  Per your comment, we could update it only in the case of 200, but in other situations it's possible you could get a different status code with different semantics, and I'd like to call out the semantics of 304 specifically.